lux-lib/src/lockfile/mod.rs

```diff
@@ -880,7 +899,19 @@ impl Lockfile<ReadOnly> {
     }
 
     /// Creates a temporary, writeable lockfile which can never flush.
-    pub fn into_temporary(self) -> Lockfile<ReadWrite> {
+    #[cfg(feature = "lua")]
+    pub(crate) fn into_temporary(self) -> Lockfile<ReadWrite> {
```

Comment:
question(rationale): I see that we're doing an override to make `into_temporary` public only to the crate here, but why on `feature = "lua"` specifically? Can't quite grasp what difference it makes :sweat_smile:

Reply:
We only need it to be `pub(crate)` for a `UserData` impl in another module.
Without the `lua` feature, this function is private to this module, because we should typically use `write_guard`.